src/imp/linux_raw/arch/outline/arm.S

```diff
@@ -0,0 +1,92 @@
+// arm syscall argument register ordering is the similar to the arm
+// userspace argument register ordering except that the syscall number
+// (nr) is passed in r7.
+//
+// arm.rs takes care of reordering the nr argument to the end for us,
+// so we only need to take care of sp by moving it to r4, r5 and r7.
+
+    .section    .text.rsix_reordered_syscall,"ax",%progbits
+    .p2align    4
+rsix_reordered_syscall:
+rsix_reordered_syscall0:
+rsix_reordered_syscall1:
+rsix_reordered_syscall2:
+rsix_reordered_syscall3:
+rsix_reordered_syscall4:
+rsix_reordered_syscall5:
+rsix_reordered_syscall6:
+    .cfi_startproc
+    push    {r4, r5, r7, lr}
+    ldr     r7, [sp, #28]
+    ldr     r4, [sp, #16]
+    ldr     r5, [sp, #20]
+    svc     #0
+    pop     {r4, r5, r7, pc}
+    .cfi_endproc
+    .size rsix_reordered_syscall, .-rsix_reordered_syscall
+
+    .section    .text.rsix_reordered_syscall1_noreturn,"ax",%progbits
+    .p2align    4
+rsix_reordered_syscall1_noreturn:
+    .cfi_startproc
+    push    {r7, lr}
+    ldr     r7, [sp, #28]
+    svc     #0
+    .inst   0xe7ffdefe
```

Comment:
As far as I understand there is "trap" mnemonic for ARM asm. I added a comment and changed the value to be a "trap" equivalent instead of a debugger breakpoint as I wrote it originally.